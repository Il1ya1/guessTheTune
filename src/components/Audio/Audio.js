import React, { Component } from 'react';
import propTypes from 'prop-types';
import classNames from 'classnames';



export default class Audio extends Component {
  static defaultProps = {
    isPlayning: false,
  }

  constructor(props) {
    super(props);
    this.audioRef = React.createRef();
    this.state = {
      isLoading: false,
      isError: false,
      isSuccess: false,
      progress: 0,
    }
  }

  componentDidMount() {

  }

  componentDidUpdate() {
    const { isPlayning } = this.props;

    if (isPlayning) {
      this.audioRef.current.play();
    }

    if (!isPlayning) {
      this.audioRef.current.pause();
    }

    this.audioRef.current.oncanplaythrough = () => this.setState({
      isLoading: true,
    });

    return null;
  }

  render() {
    const {
      src,
      isPlayning,
      startTrack,
    } = this.props;

    const playBtnClass = classNames({
      'track__button': true,
      'track__button--play': !isPlayning,
      'track__button--pause': isPlayning
    });

    return (
      <>
        <button onClick={startTrack} className={playBtnClass} type="button"></button>
        <div className="track__status">
          <audio src={src} ref={this.audioRef} />
        </div>
      </>
    )
  }

};

Audio.propTypes = {
  isLoading: propTypes.bool,
  startTrack: propTypes.func,
  progress: propTypes.number,
};