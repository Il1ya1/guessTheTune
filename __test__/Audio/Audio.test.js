import React from 'react';
import renderer from 'react-test-renderer';

import Audio from '../../src/components/Audio/Audio';

it('renders Audio', () => {
	const tree = renderer
		.create(<Audio/>)
		.toJSON();

	expect(tree).toMatchSnapshot();
});