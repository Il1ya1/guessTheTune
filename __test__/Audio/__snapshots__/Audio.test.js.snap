// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Audio 1`] = `
Array [
  <button
    className="track__button track__button--play"
    type="button"
  />,
  <div
    className="track__status"
  >
    <audio />
  </div>,
]
`;
